### PR TITLE
fix: mutationobserver memory leak

### PR DESCRIPTION
**Pull request by ravijayaramappa**
## Details
clean up strong references to observer objects on nodes that tend to live long.
observers on regular nodes will be cleaned up when the node are garbage collected

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

If yes, please describe the impact and migration path for existing applications.

## The PR fulfills these requirements:
* Have tests for the proposed changes been added? ✅
* Have you followed [these instructions](../CONTRIBUTING.md#-commit-message-conventions) to clearly describe the issue being fixed or feature enhanced? ✅

Alternative to https://github.com/salesforce/lwc/pull/1423

### CODE REVIEW

File: packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts [hunk 1]
```diff
@@ -197,8 +199,20 @@ function PatchedMutationObserver(
     return observer;
 }
 
+const sharedNodes = [document, document.head, document.body];
```
Comment (caridy) [inline packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts]: these don't always exists, we had some issues with head and body in the past, you have to be defensive I believe.

File: packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts [hunk 2]
```diff
@@ -197,8 +199,20 @@ function PatchedMutationObserver(
...
 function patchedDisconnect(this: MutationObserver): void {
     originalDisconnect.call(this);
+    // Clean up any references to the observers on the global nodes and make them eligible for GC
+    // Optimization: We clean only the global nodes that tend to last longer than other nodes.
+    forEach.call(sharedNodes, node => {
+        const observerList = node[observerLookupField];
```
Comment (caridy) [inline packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts]: this is the line with the problem... node might be undefined/null.

File: packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts (hunk 1 above)
Comment (caridy) [reply packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts]: also, imagine the case where this is evaluated in the head, while the body hasn't been rendered/parsed yet.

Comment (nolanlawson) [reply packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts]: This manual list seems inherently error-prone to me... what if an app has another node that is never removed from the DOM? E.g. `<div class="container"></div>` or similar?

**Comment by ravijayaramappa** [comment]
Closing in favor of https://github.com/salesforce/lwc/pull/1423

**Pull request closed without merging**